delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resources/ResourceCommand.scala

```diff
@@ -0,0 +1,115 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.resources
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.{CompactedJsonLd, ExpandedJsonLd}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import io.circe.Json
+
+/**
+  * Enumeration of resource commands
+  */
+sealed trait ResourceCommand extends Product with Serializable {
+
+  /**
+    * @return the project where the resource belongs to
+    */
+  def project: ProjectRef
+
+  /**
+    * @return the resource identifier
+    */
+  def id: Iri
+
+  /**
+    * @return the identity associated to this command
+    */
+  def subject: Subject
+
+}
+
+object ResourceCommand {
+
+  /**
+    * Command that signals the intent to create a new resource.
+    *
+    * @param id          the resource identifier
+    * @param project     the project where the resource belongs
+    * @param schema      the schema used to constrain the resource
+    * @param source      the representation of the resource as posted by the subject
+    * @param compacted   the compacted JSON-LD representation of the resource
+    * @param expanded    the expanded JSON-LD representation of the resource
+    * @param subject     the subject which created this event
+    */
+  final case class CreateResource(
+      id: Iri,
+      project: ProjectRef,
+      schema: ResourceRef,
+      source: Json,
+      compacted: CompactedJsonLd,
```

Comment:
Shouldn't the compacted/expanded computation append in the evaluate part ?

Reply:
Ideally yes. But since there is an edge case when the expanded and compacted has to be computed before the evaluate function, I prefered to keep them all outside.

The exception is:
You can create a resource without specifying the id. In this case, we attempt to fetch the id from the payload, meaning that you have to expand the resource **before** calling evaluate, in order to extract its id

Reply:
This is a shame because of the message sizes exchanged between the nodes... but you're right, we need to extract the id before sending the message.